core/trie/bitarray.go

```diff
@@ -0,0 +1,397 @@
+package trie
+
+import (
+	"bytes"
+	"encoding/binary"
+	"math"
+	"math/bits"
+
+	"github.com/NethermindEth/juno/core/felt"
+)
+
+const (
+	maxUint64 = uint64(math.MaxUint64)
+	bits8     = 8
+)
+
+var (
+	maxBitArray   = [4]uint64{math.MaxUint64, math.MaxUint64, math.MaxUint64, math.MaxUint64}
+	emptyBitArray = new(BitArray)
+)
+
+// BitArray is a structure that represents a bit array with length representing the number of used bits.
+// It uses a little endian representation to do bitwise operations of the words efficiently.
+// For example, if len is 10, it means that the 2^9, 2^8, ..., 2^0 bits are used.
+// The max length is 255 bits (uint8), because our use case only need up to 251 bits for a given trie key.
+// Although words can be used to represent 256 bits, we don't want to add an additional byte for the length.
+type BitArray struct {
```

Comment:
```suggestion
// Represents a bit array with length representing the number of used bits.
// It uses a little endian representation to do bitwise operations of the words efficiently.
// For example, if len is 10, it means that the 2^9, 2^8, ..., 2^0 bits are used.
// The max length is 255 bits (uint8), because our use case only need up to 251 bits for a given trie key.
// Although words can be used to represent 256 bits, we don't want to add an additional byte for the length.
type BitArray struct {
```

Just a small suggestion, I think there is no need to preface with `BitArray is a structure` since you can infer that from the definition

Reply:
Yeah, better for maintainability as well, will change it for the methods too